src/net/atlas/download.rs

```diff
@@ -303,6 +303,55 @@ impl AttachmentsDownloader {
 
         Ok(resolved_attachments)
     }
+
+    /// Check any queued attachment instances to see if we already have data for them,
+    ///  returning a vector of (instance, attachment) pairs for any of the queued attachments
+    ///  which already had the associated data
+    /// Marks any processed attachments as checked
```

Comment:
Which thread calls this function?

Reply:
This method is invoked in the thread managing the AttachmentDownloader. This is currently the P2P thread. Will add to the comment.

Reply:
Added in https://github.com/stacks-network/stacks-blockchain/pull/3086/commits/cb9f8dca0f9697e7cfe04c1b2e776de4b8aec8ce